misc/wsdl-to-ballerina/src/main/java/io/ballerina/wsdltoballerina/diagnostic/DiagnosticMessage.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.wsdltoballerina.diagnostic;
+
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.util.Objects;
+
+public class DiagnosticMessage {
```

Comment:
I think an enum would better suite this particular case. What about something like this ?
```java
public enum WsdlDiagnostic {
    WSDL_100(// add the relevant information);

    private final int code;
    private final String description;
    private final DiagnosticSeverity severity;

    WsdlDiagnostic(int code, String description, DiagnosticSeverity severity) {
        this.code = code;
        this.description = description;
        this.severity = severity;
    }

    public int getCode() {
        return code;
    }

    public String getDescription() {
        return description;
    }

    public DiagnosticSeverity getSeverity() {
        return severity;
    }
}
```

To generate the error message properly you can have a util which accepts a diagnostic-code and args.
```java
public static String constructError(WsdlDiagnostic code, Object[] args) {}
```